### ISSUE TITLE
help with truffleruby installation on macOS Sonoma

### ISSUE THREAD

**Issue by bolandross** [oracle/truffleruby#3294]
Hey folks,

even though truffleruby is not officially supported to run on macOS Sonoma, I wondered if you could help me install it on my system (MacBook Air M1, macOS Sonoma 14.0)?

Using `rbenv` I was able to easily install `truffleruby+graalvm-23.1.0`

Launching simple apps works fine. The problems only start when I want to install gems via `gem`. For example, if I start with something relatively simple, I get:

```
bolandross@MacBook-Air ~ % gem install csv
<internal:core> core/kernel.rb:234:in `gem_original_require': dlopen(/Users/bolandross/.rbenv/versions/truffleruby+graalvm-23.1.0/lib/mri/psych.bundle, 0x0002): symbol not found in flat namespace '_rb_cObject' (RuntimeError)
	from <internal:/Users/bolandross/.rbenv/versions/truffleruby+graalvm-23.1.0/lib/mri/rubygems/core_ext/kernel_require.rb>:85:in `require'
	from /Users/bolandross/.rbenv/versions/truffleruby+graalvm-23.1.0/lib/mri/psych.rb:13:in `<top (required)>'
	from <internal:core> core/kernel.rb:234:in `gem_original_require'
	from <internal:/Users/bolandross/.rbenv/versions/truffleruby+graalvm-23.1.0/lib/mri/rubygems/core_ext/kernel_require.rb>:85:in `require'
	from /Users/bolandross/.rbenv/versions/truffleruby+graalvm-23.1.0/lib/mri/rubygems.rb:608:in `load_yaml'
	from /Users/bolandross/.rbenv/versions/truffleruby+graalvm-23.1.0/lib/mri/rubygems/config_file.rb:351:in `load_file'
	from /Users/bolandross/.rbenv/versions/truffleruby+graalvm-23.1.0/lib/mri/rubygems/config_file.rb:194:in `initialize'
	from /Users/bolandross/.rbenv/versions/truffleruby+graalvm-23.1.0/lib/mri/rubygems/gem_runner.rb:71:in `do_configuration'
	from /Users/bolandross/.rbenv/versions/truffleruby+graalvm-23.1.0/lib/mri/rubygems/gem_runner.rb:33:in `run'
	from /Users/bolandross/.rbenv/versions/truffleruby+graalvm-23.1.0/bin/gem:29:in `<main>'
```

Could you give me a hint on what is missing here?

**Comment by eregon** [comment]
This is the same issue as in https://github.com/oracle/truffleruby/issues/3283.
You will need to wait for XCode 15 support in GraalVM as macOS changed things incompatibly there.

**Comment by eregon** [comment]
Let's track that in https://github.com/oracle/truffleruby/issues/3283.

**Comment by eregon** [comment]
Fixed now: https://github.com/oracle/truffleruby/issues/3283#issuecomment-1798324374

**Comment by eregon** [comment]
This fix will be in the 23.1.2 release which will be available on January 23, 2024.
